### DISCUSSION TITLE
Does Mermaid provide syntax writer API?

### DISCUSSION THREAD

**Discussion by boan-anbo** [mermaid-js/mermaid#5031] [Q&A]
I am building a 3rd party library that needs to generate Mermaid diagram definitions. 

## What I need

```ts title="Pseudo-code

interface CustomEntity {
       name: string,
       text: string,
}

const entity: CustomEntity = {
       name: "SomeEntity",
       text: "Text description of SomeEntity"
}

// I wonder if mermaid provides any APIs like this:
const mermaidSnippet = mermaid.write(entity.label, entity.text, MERMAID_ENTITY_TYPE.FLOW_CHART_NODE);

// assert.eq(mermaidSnippet, "some_entity\[\"Text description of SomeEntity\]\"")
```

Is this possible, or are we expected to implement our own solutions to spit out mermaid definitions? Thanks!

**Comment by abiriadev** [comment]
I found a useful library providing API to generate mermaid syntax programmatically: https://github.com/devinoue/type-mermaid
